### PR TITLE
stat/distuv: Improve test coverage for `Categorical` distribution + add documentation to `interfaces.go`

### PR DESCRIPTION
**Pull request by romanwerpachowski**
This change modifies the `stat/distuv` package in Gonum to:
- improve test coverage for `Categorical` distribution
- add benchmarks for `Rand` function on `Categorical`
- add documentation comments to `interfaces.go`

Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
Please start work off origin/master so that we don't have so many commits in the PR.

File: stat/distuv/interfaces.go [hunk 1]
```diff
@@ -4,19 +4,23 @@
 
 package distuv
```
Comment (kortschak) [inline stat/distuv/interfaces.go:5]: The changes in this file really should be a separate PR.

**Comment by romanwerpachowski** [comment]
> Please start work off origin/master so that we don't have so many commits in the PR.

Can you recommend a workflow? I am not terribly familiar with pull requests.

**Comment by kortschak** [comment]
When I'm working on repos that I'm not an owner of, I keep two remotes, one is the origin and the other is my fork. Before I do some work, I make sure I pull from the origin/master and then branch from there. This way, when I push to my kortschak/feature-branch remote branch, I only have the changes that I am actually making in that branch.

After the PR has been merged, I delete my feature branch, and the remote branch and pull the origin/master again, which will now contain the added feature.

With repos that I own, it's a bit easier, since I only need to have one remote, but the process is essentially still the same.

**Comment by romanwerpachowski** [comment]
Thanks.

**Comment by romanwerpachowski** [comment]
I think I messed up this branch, I'll make two new separate PRs.
